mlrun/api/db/sqldb/db.py

```diff
@@ -1859,22 +1873,25 @@ def _find_artifacts(
     ):
         """
         TODO: refactor this method
-        basically uids should be list of strings (representing uids), but we also handle two special cases (mainly for
+        basically ids should be list of strings (representing ids), but we also handle 3 special cases (mainly for
         BC until we refactor the whole artifacts API):
-        1. uids == "*" - in which we don't care about uids we just don't add any filter for this column
-        1. uids == "latest" - in which we find the relevant uid by finding the latest artifact using the updated column
+        1. ids == "*" - in which we don't care about ids we just don't add any filter for this column
+        2. ids == "latest" - in which we find the relevant uid by finding the latest artifact using the updated column
+        3. ids is a string (different than "latest") - in which the meaning is actually a uid, so we add this filter
         """
         if category and kind:
             message = "Category and Kind filters can't be given together"
             logger.warning(message, kind=kind, category=category)
             raise ValueError(message)
         labels = label_set(labels)
         query = self._query(session, Artifact, project=project)
-        if uids != "*":
-            if uids == "latest":
+        if ids != "*":
+            if ids == "latest":
                 query = self._latest_uid_filter(session, query)
+            elif isinstance(ids, str) and ids:
```

Comment:
Practically it can't happen in any flow.
But for "completeness" removed the `and ids` so empty string will get in this `if` as well, that way I'm staying with what I'm saying in the comment above (in the code):
> ids is a string..is actually a uid